src/components/SearchInput/SearchInput.jsx

```diff
@@ -1,7 +1,63 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { DebounceInput } from 'react-debounce-input';
+
+import styles from './SearchInput.module.scss';
+import SearchIcon from '../../assets/images/search-icon.png';
+import SearchLocationIcon from '../../assets/images/search-location-icon.png';
+import { NUMBER_OF_RESULTS } from '../../constants/forecaApi';
+import { dataService } from '../../services/dataService';
 
 function SearchInput() {
-  return <div>Search input</div>;
+  const [searchQuery, setSearchQuery] = useState('');
+  const [searchResults, setSearchResults] = useState([]);
+
+  useEffect(() => {
+    if (searchQuery) {
+      searchCities(searchQuery);
+    }
+  }, [searchQuery]);
+
+  async function searchCities(query) {
+    try {
+      const { locations } = await dataService.searchCity(query);
+
+      setSearchResults(locations);
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  const handleChange = e => {
+    setSearchQuery(e.target.value);
+  };
+
+  const renderSearchResults = searchResults.map(item => (
+    <li key={item.id} className={styles.searchResultItem}>
+      <img src={SearchLocationIcon} alt="location" />
+      <span>{item.name}</span>
+    </li>
+  ));
+
+  return (
+    <div className={styles.searchWrapper}>
+      <div className={styles.searchForm}>
+        <DebounceInput
```

Comment:
I want you to write it yourself, please. I allow you to use debounce function from lodash.debounce. But Debounce input should be written by yourself. Don't forget to remove unneeded packages after that

Reply:
i did it